src/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -432,13 +490,13 @@ public void SavePackageSources(IEnumerable<PackageSource> sources)
             var existingSettings = (Settings.GetSettingValues(ConfigurationContants.PackageSources, isPath: true) ??
```

Comment:
Don't you have to fix SavePackageSources method in any case. Otherwise, any changes being made to local nuget.config file, which is read-only, will be a problem, right?

With one of your changes, I believe, when you change a package source, it will change the nuget.config it comes from based on priority, instead of always writing to global nuget.config

So, in NuGet 3, it is more likely for the local nuget.config be updated compared to NuGet 2.8.6

If this cannot be fixed with this change, please file a bug to track this issue